java/code/src/com/redhat/rhn/domain/server/Server.java

```diff
@@ -1955,6 +1988,23 @@ private NetworkInterface lookupForPrimaryInterface() {
         return null;
     }
 
+    /**
+     * @param fqdnName name of the primary FQDN
+     */
+    public void setPrimaryFQDNWithName(String fqdnName) {
+        ServerFQDN newPrimaryFQDN = null;
+        for (ServerFQDN fqdn: fqdns) {
+            fqdn.setPrimary(false);
+            if (fqdn.getName().equals(fqdnName)) {
+                newPrimaryFQDN = fqdn;
+            }
+        }
+        SystemManager.storeServer(this);
```

Comment:
This `storeServer` call is needed to guarantee that we don't hit the DB unique constraint that prevents more than one primary FQDN for each server, but we don't have to call it again because DB will then be updated via SessionFilter:

https://github.com/uyuni-project/uyuni/blob/93a51e3edc32dbc7cbaccc53d8da5743aeba9176/java/code/src/com/redhat/rhn/frontend/servlets/SessionFilter.java